pkg/services/accesscontrol/resourcepermissions/store.go

```diff
@@ -675,24 +681,26 @@ func (s *store) createPermissions(sess *db.Session, roleID int64, resource, reso
 		permissions = append(permissions, p)
 	}
 
-	// If there are no missing actions for the resource (in case of access level downgrade or resource removal), we don't need to insert any prior actions
+	// If there are no missing actions for the resource (in case of access level downgrade or resource removal), we don't need to insert any actions
 	// we still want to add the action set in case of access level downgrade, but not in case of resource removal (when permission == "")
-	if len(missingActions) == 0 {
-		if s.features.IsEnabled(context.TODO(), featuremgmt.FlagAccessActionSets) && permission != "" {
-			if _, err := sess.InsertMulti(&permissions); err != nil {
-				return err
-			}
+	if len(missingActions) == 0 && permission != "" && s.features.IsEnabled(context.TODO(), featuremgmt.FlagAccessActionSets) {
+		if _, err := sess.InsertMulti(&permissions); err != nil {
+			return err
 		}
 		return nil
 	}
 
-	for action := range missingActions {
-		p := managedPermission(action, resource, resourceID, resourceAttribute)
-		p.RoleID = roleID
-		p.Created = time.Now()
-		p.Updated = time.Now()
-		p.Kind, p.Attribute, p.Identifier = p.SplitScope()
-		permissions = append(permissions, p)
+	// if we have actionsets enabled, we only want to add the action sets to the permissions table
+	// skip adding the actions to the permissions table, if we are only working with action sets
+	if !s.features.IsEnabled(context.TODO(), featuremgmt.FlagAccessActionSets) && !s.cfg.OnlyAccessActionSets {
```

Comment:
Should this be an OR not an AND? Or maybe this would be clearer:
```suggestion
	if !(s.features.IsEnabled(context.TODO(), featuremgmt.FlagAccessActionSets) && s.cfg.OnlyAccessActionSets) {
```

Reply:
So if action sets are enabled and only action sets is set, then we do not populate the fine-grained permissions.